jbpm-flow/src/main/java/org/jbpm/process/core/context/variable/CaseVariableInstance.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.core.context.variable;
+
+import java.util.Collection;
+import java.util.Objects;
+
+import org.drools.core.ClassObjectFilter;
+import org.drools.core.common.InternalKnowledgeRuntime;
+import org.jbpm.process.instance.context.variable.VariableScopeInstance;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.process.CaseData;
+import org.kie.api.runtime.rule.FactHandle;
+
+public class CaseVariableInstance<T> implements VariableInstance<T> {
+
+    private final InternalKnowledgeRuntime knowledgeRuntime;
+    private final Variable variable;
+    private final String caseFileName;
+    private final ValueReference<T> valueReference = new ValueReference<T>() {
+        @Override
+        public T get() {
+            Collection<CaseData> caseFiles = (Collection<CaseData>)
+                    knowledgeRuntime.getObjects(new ClassObjectFilter(CaseData.class));
+            if (caseFiles.size() == 1) {
+                CaseData caseFile = caseFiles.iterator().next();
+                // check if there is case file prefix and if so remove it before checking case file data
+                if (caseFile != null) {
+                    return (T) caseFile.getData(caseFileName);
+                }
+            }
```

Comment:
Shouldn’t this throw an exception? Assuming we are sure the case file should be there it should fail in case these is no case file in working memory

Reply:
I think I agree with this, although I'm pretty sure we are currently defaulting to returning `null`. But I'm all for throwing.

Reply:
let's ignore I guess this is what we do now, isn't it?